build-setup/macos.md

```diff
@@ -183,18 +183,18 @@ external processes.
 Build as follows:
 
 ```
-# TODO: add BLAS=vecLib when we upgrade to 1.9
+export BLAS=vecLib
 export BUILD_TEST=OFF
 export BUILD_CAFFE2=OFF
 export USE_NUMPY=OFF
 export USE_DISTRIBUTED=OFF
-export USE_MKLDNN=OFF
+export USE_MKLDNN=ON
```

Comment:
Turns out this doesn't work in PyTorch 1.9 when built for ARM macOS.

The fixes that make `oneDNN` work on Apple silicon are in versions 2.1.3 and 2.2 according to https://github.com/oneapi-src/oneDNN/issues/984#issuecomment-811952961.  And PyTorch 1.9 is using version 2.1.2 😿 .

So I'll make this setting conditional on the CPU type for the time being.

Reply:
It should be possible to build MKLDNN on M1 in PyTorch 1.10 - see https://github.com/pytorch/pytorch/pull/57928.  We'll have to see if there are further problems downstream of simply building it though.